src/Controller/CRUDController.php

```diff
@@ -524,7 +523,60 @@ public function batchAction(Request $request): Response
             return $this->redirectToList();
         }
 
-        return $this->$finalAction($query, $forwardedRequest);
+        return call_user_func($this->getBatchActionExecutable($action), $query, $forwardedRequest);
+    }
+
+    final protected function canBatchActionBeExecuted(string $action): bool
+    {
+        $batchActions = $this->admin->getBatchActions();
+        if (!\array_key_exists($action, $batchActions)) {
+            throw new \RuntimeException(sprintf('The `%s` batch action is not defined', $action));
+        }
+
+        $controller = $batchActions[$action]['controller'] ?? null;
+
+        if ($controller) {
+            $request = new Request();
+            $request->attributes->set('_controller', $controller);
+            try {
+                return false !== $this->container->get('controller_resolver')->getController($request);
```

Comment:
Rather than working with the controllerResolver, what do you think about creating something like a BatchActionResolver instead ? Something based on https://symfony.com/doc/current/service_container/tags.html

We could have
```
interface BatchActionInterface
{
    public function supports(AdminInterface $admin, string $action): bool;

    public function isRelevant(AdminInterface $admin, array $idx, bool $allElements, Request $request): string|bool;

    public function execute(AdminInterface $admin, ProxyQueryInterface $proxyQuery, Request $request): Response
}
```
and
```
class BatchActionResolver
{
    /** @var array<BatchActionInterface> $/
    private array $batchActions;

    public function __construct(array $batchActions)
    {
         $this->batchActions = $batchActions;
    }

    public function get(AdminInterface $adminInterface, string $action): BatchActionInterface
    {
        foreach ($this->batchActions as $batchAction) {
             if ($batchAction->supports($admin, $action) {
                 return $batchAction;
             }
        }
    }
}
```

And then the controller would do
```
$this->container->get('batch_action_resolver')->get($this->admin, $action);
```
And you don't even need to add a new `controller` key.

Reply:
From my point of view this is overcomplexify something that is a concept already know by Symfony developer (create a controller) and will also support any feature related to a controller. It would need to test but service injection in controller method for example.

Also the BatchActionResorlver and BatchActionInterface the way it have been implement would limit one action per class. This might be something that people wish but Using the controller approach would give the choice to people to do what they want.

It will also load a bunch of service and executing some logic to find one (that need to be implemented) on top of having to configure priority (what if multiple action support it ?).

And the controller approach is easy to use directly with the AdminExtensionInterface, nothing to add there.

Also it's a lot easier to document. It could be added to the current documentation with a couple of lines.

Reply:
I would add that people might then to extends the class from AbstractController or event CRUDController for some reason, they are most likely register it as a controller anyway.

Reply:
> From my point of view this is overcomplexify something that is a concept already know by Symfony developer (create a controller) and will also support any feature related to a controller. It would need to test but service injection in controller method for example.
> 
> Also the BatchActionResorlver and BatchActionInterface the way it have been implement would limit one action per class. This might be something that people wish but Using the controller approach would give the choice to people to do what they want.
> 
> It will also load a bunch of service and executing some logic to find one (that need to be implemented) on top of having to configure priority (what if multiple action support it ?).
> 
> And the controller approach is easy to use directly with the AdminExtensionInterface, nothing to add there.
> 
> Also it's a lot easier to document. It could be added to the current documentation with a couple of lines.

I wanted to have an interface to avoid relying on the name like it was before
```
$finalAction = sprintf('batchAction%s', $camelizedAction);
```
but I'm not familiar with the `_controller` param. Do you mean you're passing the controller method and not only the controller name ?
```
'_controller' => 'App\Controller\AdminController::batchClone',
```
and not 
```
'_controller' => 'App\Controller\AdminController',
```
it could be enough then.

Then. rather than your if/else strategy, can't we do
```
$camelizedAction = InflectorFactory::create()->build()->classify($action);
$finalAction = sprintf('batchAction%s', $camelizedAction);

$controller = $batchActions[$action]['controller'] ?? sprintf('%s::%s', self::class, $finalAction);

$request = new Request();
$request->attributes->set('_controller', $controller);
try {
    return false !== $this->container->get('controller_resolver')->getController($request);
} catch (\Throwable $error) {
     return false;
}
```
dunno if it's a good idea.

Also, in case they are extra values in the request, should we prefer `$request = clone $existingRequest` ?

Reply:
We might also want to deprecate the magic behind the `batchAction%s` name ; and prefer people to pass 
```
'_controller' => 'App\Controller\CRUDController::batchActionFoo',
```
every time to stop with variable calls.

Reply:
Yes the controller is not just the class but the full path with function like you would define a specific path to a controller in a routing configuration. If you just specify the class in imply __invoke (I think).

I was keeping the finalAction action logic to be backward compatible. But I could try your way to implement it.